### PR TITLE
BaseVault multiple fixes

### PR DESCRIPTION
**Pull request by DaigaroCota**
This PR addresses:

- the issue with the first deposit small amount that leads to DOS #214 
- add version to vaults and a setter function for vault's ratings in the Chief.sol #224 
- out of date provider when blindly calling `setActiveProvider()` #192

### CODE REVIEW

**Comment by DaigaroCota** [comment]
Self note:
Need to understand why these are failing:
This is a problem with Foundry most likely.
---------------------------------------------
Failing tests:

Encountered 1 failing test in test/forking/mainnet/CompoundV3.t.sol:CompoundV3ForkingTest
[FAIL. Reason: Setup failed: Contract 0x5991A2dF15A8F6A256D3Ec51E99254Cd3fb576A9 does not exist and is not marked as persistent, see `vm.makePersistent()`] setUp() (gas: 0)

Encountered 1 failing test in test/forking/polygon/DForcePolygon.t.sol:DForcePolygonForkingTest
[FAIL. Reason: Setup failed: Revert] setUp() (gas: 0)

Encountered 1 failing test in test/forking/polygon/FlasherAaveV3.t.sol:FlasherAaveV3ForkingTest
[FAIL. Reason: Setup failed: Revert] setUp() (gas: 0)

Encountered 1 failing test in test/forking/polygon/WePiggyPolygon.t.sol:WePiggyPolygonForkingTest
[FAIL. Reason: Setup failed: Revert] setUp() (gas: 0)

**Comment by DaigaroCota** [comment]
@brozorec this is ready for review. Please see notes above.

**Comment by brozorec** [comment]
Hey @DaigaroCota I just had a look at this PR. I think I understand why you brought back the `debtAsset` in the BorrowingVault. It's because it's needed in `setProviders`, right? If this is the only reason, I might suggest we turn `_setProviders(providers)` into an abstract method in BaseVault. That will allow the BorrowingVault to implement it by using both `_asset` and `_debtAsset` w/o checking nullity and the YieldVault only using `_asset`.

**Comment by brozorec** [comment]
Regarding the ratings, I suggest we switch from a string-based rating to a number-based one (1-100). That's easier to handle and far more scalable, at least from an SC perspective. 

How this number gets interpreted on the client side is another question. We can display whatever we want: "A+", "A-", ... or 10-base or 20-base. A rating with a range 1-100 gives us enough flexibility.

**Comment by DaigaroCota** [comment]
> Regarding the ratings, I suggest we switch from a string-based rating to a number-based one (1-100). That's easier to handle and far more scalable, at least from an SC perspective.
> 
> How this number gets interpreted on the client side is another question. We can display whatever we want: "A+", "A-", ... or 10-base or 20-base. A rating with a range 1-100 gives us enough flexibility.

Agree.  I will simply assign it a uint256, and restrict range [1,100]

**Comment by DaigaroCota** [comment]
@brozorec, this should be ready to merge once approved.

File: packages/protocol/src/Chief.sol
Comment (brozorec) [inline packages/protocol/src/Chief.sol:92]: Inconsistent naming, why do we use here "_name"? There are no global `factory` or `_deployData`

File: packages/protocol/src/Chief.sol
Comment (brozorec) [inline packages/protocol/src/Chief.sol:120]: In "deployVault" above, the naming is " underscore + name" while here it becomes "name + underscore"

File: packages/protocol/src/Chief.sol
Comment (brozorec) [inline packages/protocol/src/Chief.sol:256]: some comment as above

**Review by brozorec — CHANGES_REQUESTED** [review]

File: packages/protocol/src/abstracts/BaseVault.sol
Comment (brozorec) [inline packages/protocol/src/abstracts/BaseVault.sol:51]: Why not just "0.0.0"?

It's a bit weird all zeroed; maybe we start with "0.0.1"?

File: packages/protocol/src/Chief.sol
Comment (DaigaroCota) [reply packages/protocol/src/Chief.sol]: Updated variable naming with no underscore since you are right; there is no variable collision.

File: packages/protocol/src/Chief.sol
Comment (DaigaroCota) [reply packages/protocol/src/Chief.sol]: Updated.

File: packages/protocol/src/abstracts/BaseVault.sol
Comment (DaigaroCota) [reply packages/protocol/src/abstracts/BaseVault.sol]: updated.

**Comment by DaigaroCota** [comment]
@brozorec review comments have been addressed.

**Review by brozorec — APPROVED** [review]

File: packages/protocol/test/forking/arbitrum/AaveV3.t.sol
Comment (pedrovalido) [inline packages/protocol/test/forking/arbitrum/AaveV3.t.sol:24]: Why do we have vm.makePersistent in this test and not the other ones?

Comment (DaigaroCota) [reply packages/protocol/test/forking/arbitrum/AaveV3.t.sol:24]: > Why do we have vm.makePersistent in this test and not the other ones? @DaigaroCota

@pedrovalido I forgot to remove it. Please do so in one of your provider implementations. It should not be affecting test passing.

Comment (pedrovalido) [reply packages/protocol/test/forking/arbitrum/AaveV3.t.sol:24]: Got it will do. It doesnt affect, I was just curious 😅

**Pull request merged**
